src/hooks/common.rs

```diff
@@ -217,7 +217,7 @@ pub trait ImguiRenderLoop {
 
     /// If this function returns true, the WndProc function will not call the
     /// procedure of the parent window.
-    fn should_block_messages(&self, _io: &Io) -> bool {
+    fn should_block_messages(&self, _io: &mut Io) -> bool {
```

Comment:
This is a breaking API change which I'd strongly prefer _not_ having, for a number of reasons:
- Clients already rely on `io` being immutable; changing this would cause a number of compilation issues on projects depending on `hudhook`, and this is bad unless properly communicated and planned for.
- `should_block_messages` sounds like a question asked of the current state; it is not immediately clear to the user why it would mutate state anywhere.
- There are no strongly compelling reasons for such a big behavior change; the client code (examples in this case) should rather adapt.